source/state_representation/README.md

```diff
@@ -159,6 +159,53 @@ StateRepresentation::CartesianTwist wVa("a", Eigen::Vector3d(1, 0, 0));
 StateRepresentation::CartesianPose wPa = period * wVa; // note that wVa * period is also implemented
 ```
 
+### Cartesian state distance and norms
+
+As a `CartesianState` represents a spatial transformation, distance between states and norms computations have been
+implemented. The distance functions is represented as the sum of the distance over all the state variables:
+
+```cpp
+using namespace StateRepresentation;
+CartesianState cs1 = CartesianState::Random("test");
+CartesianState cs2 = CartesianState::Random("test");
+
+double d = cs1.dist(cs2);
+// alternatively one can use the friend type notation
+d = dist(cs1, cs2)
+```
+
+By default, the distance is computed over all the state variables.
+One can specify the state variable to consider using the `CartesianStateVariable` enumeration:
+
+```cpp
+// only the distance in position
+double d_pos = cs1.dist(cs2, CartesianStateVariable::POSITION);
+// distance over the wrench
+double d_wrench = cs1.dist(cs2, CartesianStateVariable::WRENCH);
+```
+
+The norm of a state is using a similar API and returns the norms of each state variable:
+
+```cpp
+using namespace StateRepresentation;
+CartesianState cs = CartesianState::Random("test");
+// default is norm over all the state variables, hence vector is of size 8
+std::vector<double> norms = cs.norms();
+// for only norm over the pose you need to specify it
+std::vector<double> pose_norms = cs.norms(CartesianStateVariable::POSE);
```

Comment:
That is more a c++ problem. You can't have a function with the same signature returning both a `std::vector<double>` or a `double`. Only way is to have a different signature (name and parameters). Although I have to check maybe with new version of c++ there are some tricks to do it.

Reply:
As discussed, I can add the `norm` function (no s) and check that the `CartestianStateVariable` as input is one of the single state variables (`POSITION`, `ORIENTATION`, ...). Happy to do it if it brings value.

Reply:
> That is more a c++ problem. You can't have a function with the same signature returning both a `std::vector<double>` or a `double`. Only way is to have a different signature (name and parameters). Although I have to check maybe with new version of c++ there are some tricks to do it.

`std::variant` is the ideal pattern for this kind of thing, it's really nice to use once you get familiar with it. But for simplicity on a user side I agree a `norm()` function would be a bit more straightforward. Although, just as a consideration, I think for dominic's example it's not such a leap for the user to do `double vel_norm = cs.get_linear_velocity.norm()`, same with other sub-components of state. I don't really think this needs an extra wrapper.

Reply:
Yes but then how do you treat the linear and angular space of the velocity? Do you do the norm of the velocity vector (linear and angular) concatenated? Or you sum the two norms?

Reply:
The only reason I suggest it is because I know there will be a whole new switch case for such a norm function, which bloats the code. I personally don't mind the difference too much between:
`cs.norm(CartestianStateVariable::TWIST)` vs. 
`cs.get_linear_velocity.norm() + cs.get_angular_velocity.norm()`
But, if you do want to add the `double norm()` function I also won't complain.

Reply:
Ah but of course it won't even need more switch cases if just wraps the `norms()` function and extracts `.front()`.

Reply:
I actually misread your comments. Yes I also agree that `cs.get_linear_velocity.norm()` (and same for other subspaces) would be the easiest way to go. There is also the possibility to use the `norms` function but although there won't be a switch there will be a checking that the user is actually asking for one of the subspace, otherwise we end up with my problem on what to do with `twist` in your example.

Reply:
Ah yeah totally, I forgot about `cs.get_linear_velocity.norm()`. Ignore my comment then